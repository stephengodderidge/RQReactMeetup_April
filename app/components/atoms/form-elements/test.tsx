// import React from 'react';
// import { render, fireEvent } from 'react-testing-library';
// import { Checkbox, RadioButton } from './';

// const onClick = jest.fn();

// describe('Form Elements', () => {
//   afterEach(() => {
//     onClick.mockClear();
//   });
//   // #region Checkbox
//   describe('Checkbox', () => {
//     test('Renders', () => {
//       const { container } = render(<Checkbox onChange={onClick} value="" />);
//       expect(container).toMatchSnapshot();
//     });
//     test('Handles Click', () => {
//       const testId = 'checkbox';
//       const { getByTestId } = render(
//         <Checkbox onChange={onClick} value="" data-testid={testId} />,
//       );
//       expect((getByTestId(testId) as HTMLInputElement).checked).toBe(false);
//       fireEvent.click(getByTestId(testId));
//       expect(onClick).toHaveBeenCalledTimes(1);
//       expect((getByTestId(testId) as HTMLInputElement).checked).toBe(true);
//     });
//   });
//   // #endregion Checkbox
//   // #region Radio Button
//   describe('Radio Button', () => {
//     test('Renders', () => {
//       const { container } = render(<RadioButton onChange={onClick} value="" />);
//       expect(container).toMatchSnapshot();
//     });
//     test('Handles Click', () => {
//       const testId = 'radio-button';
//       const { getByTestId } = render(
//         <RadioButton onChange={onClick} value="" data-testid={testId} />,
//       );
//       expect((getByTestId(testId) as HTMLInputElement).checked).toBe(false);
//       fireEvent.click(getByTestId(testId));
//       expect(onClick).toHaveBeenCalledTimes(1);
//       expect((getByTestId(testId) as HTMLInputElement).checked).toBe(true);
//     });
//   });
//   // #endregion Radio Button
// });
