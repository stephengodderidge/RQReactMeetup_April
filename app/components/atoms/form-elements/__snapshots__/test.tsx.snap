// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Elements Checkbox Renders 1`] = `
<div>
  <span
    class="MuiButtonBase-root-17 MuiIconButton-root-11 MuiPrivateSwitchBase-root-7 MuiCheckbox-root-1 MuiCheckbox-colorPrimary-5"
  >
    <span
      class="MuiIconButton-label-16"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-20"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        class="MuiPrivateSwitchBase-input-10"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-29"
    />
  </span>
</div>
`;

exports[`Form Elements Radio Button Renders 1`] = `
<div>
  <span
    class="MuiButtonBase-root-17 MuiIconButton-root-11 MuiPrivateSwitchBase-root-7 MuiRadio-root-36 MuiRadio-colorPrimary-39"
  >
    <span
      class="MuiIconButton-label-16"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-20"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        class="MuiPrivateSwitchBase-input-10"
        type="radio"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-29"
    />
  </span>
</div>
`;
